Added some security best practices and key rotation info (#253)
DevCycleHQ/devcycle-docs: docs/best-practices/security/api-keys-secret-scanning.md

To maintain the security of your API and SDK keys, follow these best practices:

1. **Store your keys securely**: Keep your API and SDK keys in a secure location, such as environment variables, secret managers, or secure file storage. Avoid hardcoding keys directly into your source code or configuration files.
2. **Limit key access**: Restrict access to your API and SDK keys only to the necessary personnel and services.
3. **Regularly review key usage**: Periodically review the usage of your API and SDK keys to detect any anomalies or unauthorized access.
4. **Rotate keys**: Rotate your API and SDK keys periodically, especially when team members leave the project or in case of a security breach as mentioned above.
5. **Use different keys for different environments**: Use separate API and SDK keys for different environments (e.g., development, staging, and production) to minimize the potential impact of a compromised key. [Luckily, this is made very easy within DevCycle](/home/feature-management/organizing-your-flags-and-variables/environments)

### Understanding SDK Key Types and their security implications

DevCycle splits its SDK keys into 3 categories:

> 1. **Client**: Meant for single-user contexts, these keys have a persistent data store for the user while the SDK is being used. Evaluation of Features happens directly on the client's device. These SDKs also include the Mobile SDKs. For more information on the difference between mobile and standard client-side SDK keys, read [API and SDK keys](/home/feature-management/organizing-your-flags-and-variables/api-and-sdk-keys).

Reviewer: ```suggestion
1. **Client**: Meant for single-user contexts, these keys are used in SDK's that have a persistent data store for the user while the SDK is being used. Evaluation of Features happens in the Client SDK API Worker, which then returns all valid features/variables for a given user. These SDKs also include the Mobile SDKs. For more information on the difference between mobile and standard client-side SDK keys, read [API and SDK keys](/home/feature-management/organizing-your-flags-and-variables/api-and-sdk-keys).
These keys are considered safe to be in a client facing application - but should not be checked into source control.
```

> 2. **Mobile**: These keys are used for Mobile SDKs and provide read-only access to the features accessible by a given user on your DevCycle environment. They grant access to the DevCycle SDK API, which returns user-customized configurations including feature information that they are permitted to access.

Reviewer: ```suggestion
2. **Mobile**: These keys are used for Mobile SDKs and provide read-only access to the features accessible by a given user on your DevCycle environment. They grant access to the DevCycle SDK API, which returns user-customized configurations including feature information that they are permitted to access. These keys are the same degree of severity as the Client SDK Keys - but should be considered safe to be in built mobile application binaries, but not checked into source control.
```

> 3. **Server**: Used for server-side SDKs, these keys provide read-only access to the features on your DevCycle environment. They are used for SDKs that make continuous calls to the DevCycle APIs for each SDK interaction per user and have access to the full project configuration data.

Reviewer: ```suggestion
3. **Server**: Used for server-side SDKs, these keys provide read-only access to the features on your DevCycle environment. They are used for SDKs that make continuous calls to the DevCycle APIs for each SDK interaction per user and have access to the full project configuration data. These are considered the highest security keys, and should be immediately rotated if compromised.
```
